cache_test.go

```diff
@@ -599,6 +599,47 @@ func Test_Cache_Delete(t *testing.T) {
 	assert.NotContains(t, cache.items.values, "1")
 }
 
+func Test_Cache_GetOrSet(t *testing.T) {
+	cache := prepCache(time.Hour)
+	item, retrieved := cache.GetOrSet("test", "1", WithTTL[string, string](time.Minute))
+	require.NotNil(t, item)
+	assert.Same(t, item, cache.items.values["test"].Value)
+	assert.False(t, retrieved)
+
+	item, retrieved = cache.GetOrSet("test", "1", WithTTL[string, string](time.Minute))
+	require.NotNil(t, item)
+	assert.Same(t, item, cache.items.values["test"].Value)
+	assert.True(t, retrieved)
+
+	item, retrieved = cache.GetOrSet("test2", "1", WithTTL[string, string](time.Microsecond))
+	require.NotNil(t, item)
+	assert.Same(t, item, cache.items.values["test2"].Value)
+	assert.False(t, retrieved)
+
+	time.Sleep(time.Millisecond)
+	item, retrieved = cache.GetOrSet("test2", "2", WithTTL[string, string](time.Minute))
+	require.NotNil(t, item)
+	assert.Same(t, item, cache.items.values["test2"].Value)
+	assert.False(t, retrieved)
+}
+
+func Test_Cache_GetAndDelete(t *testing.T) {
+	cache := prepCache(time.Hour, "test1", "test2", "test3")
+	item := cache.Set("test1", "value123", time.Minute)
```

Comment:
To have proper test isolation, it would be best if we didn't use methods that are being tested elsewhere here. Since you already insert `test1` in `prepCache`, you should be able to extract that item by using `cache.items.lru.Front()`